services/ui-src/src/components/fields/DateField.tsx

```diff
@@ -14,16 +14,20 @@ export const DateField = ({ name, label, sxOverride, ...props }: Props) => {
   const form = useFormContext();
   form.register(name);
 
-  const [displayValue, setDisplayValue] = useState("");
+  const [displayValue, setDisplayValue] = useState<string>("");
+  const [formattedValue, setFormattedValue] = useState<string>("");
```

Comment:
tracking these separately so we can access the formatted valued in `onBlur`, which is necessary so we can check for incomplete dates `onBlur` and tell users the entered date is invalid.